internal/moduletest/graph/transform_state_cleanup.go

```diff
@@ -50,14 +54,13 @@ func (t *TestStateCleanupTransformer) Transform(g *terraform.Graph) error {
 
 	// connect all cleanup nodes in reverse-sequential order of run index to
 	// preserve existing behavior, starting from the root cleanup node.
-	// TODO: Parallelize cleanup nodes execution instead of sequential.
 	added := make(map[string]bool)
 	var prev dag.Vertex
 	for _, v := range slices.Backward(t.opts.File.Runs) {
 		key := v.GetStateKey()
 		if _, exists := added[key]; !exists {
 			node := cleanupMap[key]
-			if prev != nil {
+			if prev != nil && !node.parallel {
```

Comment:
this feels too simple to me? what if you have a test file with a run block a that must happen first then two run blocks b and c that can run in parallel and they all have independent states. I'd expect b and c to be cleaned up in parallel but for a to still wait for both of them to finish.

Can you recreate the edges that exist between all the nodes for each state key, and then apply them to the cleanup nodes or something?